### ISSUE TITLE
_cat indices with pattern does not list closed indices

### ISSUE THREAD

**Issue by martinhynar** [elastic/elasticsearch#10242]
I am on ES 1.4.4 and I want to list indices with some prefix

```
curl -XPUT 'localhost:9200/quick'
curl -XPOST 'localhost:9200/quick/_close'
```

Then, there is difference between querying exact index and using prefix with wildcard
```
curl 'localhost:9200/_cat/indices/quick'
close quick

curl 'localhost:9200/_cat/indices/quic*'
  --- nothing ---
```

On open index, the output is identical.

**Comment by GlenRSmith** [comment]
:+1:

**Comment by javanna** [comment]
I confirm, referring to an index explicitly makes it show up, while when using wildcards it all depends on the indices options (expand_wildcards etc.) which are hardcoded in the cat api though. The problem is [here](https://github.com/elastic/elasticsearch/blob/master/src/main/java/org/elasticsearch/rest/action/cat/RestIndicesAction.java#L75), you can never get back closed indices through cat indices unless you refer to them explicitly. @drewr can you comment on this? This is probably the only cat api where showing closed indices might make sense, then we might want to change how we resolve wildcards internally?

**Comment by javanna** [comment]
ping @drew @dakrone what do you think?

**Comment by javanna** [comment]
This was solved by #18545 in version 2.3.4 and 5.0.0-alpha3.
